environment.sh

```diff
@@ -36,52 +35,101 @@ done
 # Parse environment name
 env_name=$(basename "`pwd`")
 env_name+="_$env_type"
+branch_name=$(git rev-parse --abbrev-ref HEAD)
+# Determine which requirements.txt to install from
+if [ $env_type == 'simulation' ]; then
+  install_file="requirements.txt"
+elif [ $env_type == 'artifact' ]; then
+  install_file="artifact_requirements.txt"
+else
+  echo "Invalid environment type. Valid argument types are 'simulation' and 'artifact'."
+  return 
+fi
+requirements_modification_time="$(date -r $install_file '+%Y-%m-%d %H:%M:%S')"
+
+# Pull repo to get latest changes from remote if remote exists
+git ls-remote --exit-code --heads origin $branch_name >/dev/null 2>&1
```

Comment:
Don't we need to pull before checking the requirements modification time? The requirements file could have been updated on the remote.